python/activity_stream/activity_stream.py

```diff
@@ -694,6 +694,20 @@ def rescan(self, force_activity_stream_update=False):
             
     def note_update(self,entity, note_id):       
         note_thread_data = self._data_manager.get_note(note_id)
+        if not note_thread_data:
+            self._bundle.log_debug("Cannot update note cache for note %s - Note is not cached." % note_id)
+
+        # Remove any previous __note_thumbnail__
+        note_thread_data_cpy = list(note_thread_data)
+        current_attachments = []
+        for attachment in note_thread_data_cpy:
+            if "this_file" in attachment:
+                file_name = attachment["this_file"]["name"]
+                if re.search("__note_thumbnail__", file_name):
```

Comment:
[minor] You can also just do `"__note_thumbnail__" in file_name` since strings are iterable.